urllib3/connectionpool.py

```diff
@@ -666,6 +666,10 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                                 release_conn=release_conn, body_pos=body_pos,
                                 **response_kw)
 
+        def drain_and_release_conn(response):
+            response.read()
```

Comment:
Maybe yes.

In the case of a redirect, the entire response has probably already been read off the wire and is sitting in a buffer.  And this is probably likely for many of the non-200 cases as well?

I'm going to push this to my production systems and see if I find any exceptions in this.